io.quarkus.it.smallrye.config.AppConfigTest failing on Mac M1 (ordering issue?)

### Describe the bug

It looks like the comparison in the test in io.quarkus.it.smallrye.config.AppConfigTest is too brittle, and it's started failing on M1. I have disabled the test as part of https://github.com/quarkusio/quarkus/pull/27156. This issue tracks fixing the test and then re-enabling it. 

```
2022-09-19T10:01:23.2575320Z [ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.017 s <<< FAILURE! - in io.quarkus.it.smallrye.config.AppConfigTest
2022-09-19T10:01:23.2575750Z [ERROR] io.quarkus.it.smallrye.config.AppConfigTest.getToString  Time elapsed: 0.014 s  <<< FAILURE!
2022-09-19T10:01:23.2575980Z java.lang.AssertionError: 
2022-09-19T10:01:23.2576120Z 1 expectation failed.
2022-09-19T10:01:23.2576310Z Response body doesn't match expectation.
2022-09-19T10:01:23.2576500Z Expected: is "AppConfig{name=app, info=Info{count=10, alias=alias}}"
2022-09-19T10:01:23.2576710Z   Actual: AppConfig{name=app, info=Info{alias=alias, count=10}}
```

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/integration-tests/smallrye-config/src/test/java/io/quarkus/it/smallrye/config/AppConfigTest.java
+++ b/integration-tests/smallrye-config/src/test/java/io/quarkus/it/smallrye/config/AppConfigTest.java
@@ -2,7 +2,9 @@ package io.quarkus.it.smallrye.config;
 
 import static io.restassured.RestAssured.given;
 import static javax.ws.rs.core.Response.Status.OK;
-import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.endsWith;
+import static org.hamcrest.Matchers.startsWith;
 
 import org.junit.jupiter.api.Test;
 
@@ -16,6 +18,10 @@ public class AppConfigTest {
                 .get("/app-config/toString")
                 .then()
                 .statusCode(OK.getStatusCode())
-                .body(is("AppConfig{name=app, info=Info{count=10, alias=alias}}"));
+                .body(startsWith("AppConfig{"),
+                        containsString("name=app"),
+                        containsString("count=10"),
+                        containsString("alias=alias"),
+                        endsWith("}}"));
     }
 }

